File: Backend_DotNet/WheelShare_dotNet/WheelShare_dotNet/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WheelShare_dotNet.Models
{
    [Table("Vehicles")]
    public class Vehicle
    {

        [Key]
        public int VehicleId { get; set; }

        [Required]
        public int DriverId { get; set; }

        [Required, MaxLength(20)]
        public string VehicleType { get; set; }
        // Mini | Sedan | SUV

        [Required, MaxLength(20)]
        public string VehicleNumber { get; set; }

        public int Seats { get; set; }

        public bool IsActive { get; set; } = true;

        [ForeignKey("DriverId")]
        public Driver Driver { get; set; }
    }
}

File: Backend_Springboot/cab-booking/src/main/java/com/wheelshare/cabbooking/entity/Vehicle.java
package com.wheelshare.cabbooking.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "Vehicles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer vehicleId;

    @Column(nullable = false)
    private Integer driverId;

    @Column(nullable = false, length = 20)
    private String vehicleType;

    @Column(nullable = false, length = 20)
    private String vehicleNumber;

    private Integer seats;

    private Boolean isActive = true;

    @ManyToOne
    @JoinColumn(name = "DriverId", insertable = false, updatable = false)
    private Driver driver;
}
